app/v1/connectors/BaseDesConnector.scala

```diff
@@ -36,54 +36,59 @@ trait BaseDesConnector {
 
   def post[Body: Writes, Resp](body: Body, uri: DesUri[Resp])(implicit ec: ExecutionContext,
                                                               hc: HeaderCarrier,
-                                                              httpReads: HttpReads[DesOutcome[Resp]]): Future[DesOutcome[Resp]] = {
+                                                              httpReads: HttpReads[DesOutcome[Resp]],
+                                                              correlationId: String): Future[DesOutcome[Resp]] = {
 
     def doPost(implicit hc: HeaderCarrier): Future[DesOutcome[Resp]] = {
       http.POST(s"${appConfig.desBaseUrl}/${uri.value}", body)
     }
 
-    doPost(desHeaderCarrier(hc))
+    doPost(desHeaderCarrier(hc.withExtraHeaders("CorrelationId" -> correlationId)))
```

Comment:
Can we add the addition to the header into the desHeaderCarrier please, rather than creating another header carrier with .withExtraheaders